src/content/docs/controller.md

```diff
@@ -2,4 +2,218 @@
+
+    // Now you can use the state of the controller:
+    if state.button_a.is_pressed() {
+      // Do something repeatedly when the A button is pressed
+    } else {
+      // Do something else repeatedly when the A button is not pressed
+    }
+
+    sleep(Controller::UPDATE_INTERVAL).await;
+    // Use the state of the controller here
+  }
+}
+```
+
+Button states can be checked using:
+
+-   `button_a`/`button_b`/`button_x`/`button_y`
+-   `button_up`/`button_down`/`button_left`/`button_right`
+-   `button_l1`/`button_l2`/`button_r1`/`button_r2`
+-   `button_power` (note that while you can use this in your program, holding this button down _will_ stop the program (and eventually turn off the controller), which you probably don't want to happen in a match.)
+
+Joystick states can be checked using:
+
+-   `left_stick` (for the left joystick)
+-   `right_stick` (for the right joystick)
+
+## Button states
+
+To check whether a button is _currently pressed down_, you can use the `is_pressed` method on the button. This method returns `true` if the button is pressed and `false` otherwise.
+
+> Wait, but I want to know when a button was _just_ pressed!
+
+Don't worry! In VEXCode, you may have had to declare a separate variable just to keep track of the previous state of a button. In vexide, we do this for you!
+
+You can use the `is_now_pressed` method to check if a button was just pressed. This method returns `true` if the button was pressed in the last update (i.e., it was released the last time `Controller::state` was called and is now pressed) and `false` otherwise. This is quite useful for implementing actions like toggling a pneumatic piston or changing the state of a subsystem.
+
+> [!NOTE]
+> If a button was quickly pressed and released in between two updates, the press will not be detected by `is_now_pressed` and will be dropped. Additionally, the controller's update interval is 25ms, so the minimum time between two updates is 25ms.
+
+```rs
+// @fold start
+#![no_std]
+#![no_main]
+
+use vexide::prelude::*;
+
+// @fold end
+#[vexide::main]
+async fn main(peripherals: Peripherals) {
+  let mut controller = peripherals.primary_controller;
+  let mut piston = DigitalOut::new(peripherals.port1);
+
+  loop {
+    let state = controller.state().expect("Failed to get controller state");
+    //                             ^
+    // [We use `expect` here, but in production, you should handle the error more gracefully]
+
+    //                (              )
+    if state.button_a.is_now_pressed() {
+    //                ^
+    //         [Returns true if the A button was just pressed]
+      piston.toggle();
+    }
+
+    sleep(Controller::UPDATE_INTERVAL).await;
+  }
+}
+```
+
+## Joystick states
+
+The VEX controller has two joysticks: one on the left and one on the right. You can access the state of these joysticks using the `left_stick` and `right_stick` methods on the `ControllerState` struct, respectively.
```

Comment:
suggestion: methods -> fields (or properties?)

Reply:
Yeah, I keep forgetting which things are getters and which things are `pub` properties/attributes